src/lamp_py/ingestion/utils.py

```diff
@@ -226,55 +226,63 @@ def hash_gtfs_rt_row(row: Any) -> Tuple[bytes]:
 
 def hash_gtfs_rt_table(table: pyarrow.Table) -> pyarrow.Table:
     """
-    add lamp_records_hash column to pyarrow table, if not already present
+    add GTFS_RT_HASH_COL column to pyarrow table, if not already present
     """
-    table_columns = table.column_names
-    if GTFS_RT_HASH_COL in table_columns:
+    hash_columns = table.column_names
+    if GTFS_RT_HASH_COL in hash_columns:
         return table
 
-    table_columns.remove("feed_timestamp")
-    table_columns = sorted(table_columns)
+    hash_columns.remove("feed_timestamp")
+    hash_columns = sorted(hash_columns)
+
+    hash_schema = table.schema.append(
+        pyarrow.field(GTFS_RT_HASH_COL, pyarrow.large_binary())
+    )
 
     table = pl.from_arrow(table)
 
-    return table.with_columns(
-        table.select(table_columns)
-        .map_rows(hash_gtfs_rt_row, return_dtype=pl.Binary)
-        .to_series(0)
-        .alias(GTFS_RT_HASH_COL)
-    ).to_arrow()
+    return (
+        table.with_columns(
+            table.select(hash_columns)
+            .map_rows(hash_gtfs_rt_row, return_dtype=pl.Binary)
+            .to_series(0)
+            .alias(GTFS_RT_HASH_COL)
+        )
+        .to_arrow()
+        .cast(hash_schema)
```

Comment:
This `cast` is the only real applicable change here, the rest is hopefully making variable names clearer